webapp/src/js/controllers/contacts.js

```diff
@@ -430,16 +434,35 @@ var _ = require('underscore'),
     var changeListener = Changes({
       key: 'contacts-list',
       callback: function(change) {
-        var limit = liveList.count();
+        const limit = liveList.count();
         if (change.deleted && change.doc.type !== 'data_record') {
           liveList.remove(change.doc);
         }
 
-        var withIds =
+        if (change.doc) {
+          liveList.invalidateCache(change.doc._id);
+
+          // Invalidate the contact for changing reports with visited_contact_uuid
+          if (change.doc.fields) {
+            liveList.invalidateCache(change.doc.fields.visited_contact_uuid);
+          }
+
+          // Invalidate the parent to handle changing primary contacts
+          if (change.doc.type === 'person' && change.doc.parent) {
```

Comment:
Unfortunately, there is no constraint that the primary contact of a place must belong to that place.
Is this something we support or exists in the field? cc @abbyad @garethbowen

Reply:
We do support primary contacts not being in the place, although I don't know that it is used or needed.

Reply:
After conversation with design folk, looks like I'll be removing primary contacts from LHS next week.  If you're cool with pushing as-is -- I'll remove this block completely in a PR next week.